frontend/ui/uimanager.lua

```diff
@@ -1552,21 +1552,30 @@ function UIManager:suspend()
     end
 end
 
-function UIManager:reboot()
+function UIManager:askForReboot(message_text)
     -- Should always exist, as defined in `generic/device` or overwritten with `setEventHandlers`
     if self.event_handlers.Reboot then
         -- Give the other event handlers a chance to be executed.
         -- 'Reboot' event will be sent by the handler
-        UIManager:nextTick(self.event_handlers.Reboot)
+        UIManager:nextTick(self.event_handlers.Reboot, message_text)
```

Comment:
Not new, but that syntax means the function cannot rely on `self`` (which is currently doesn't).